### PR TITLE
Milestone 2 Code Review Request

### PR DESCRIPTION
**Pull request by JelilFaisalAbudu**
- This milestone have been completed and ready for your review.
- The user interface has a welcome message for the player.
- The user is given the option to choose the character/symbol to use in the play.
- There is also a test  to switch between each player.
- All methods to be used to run the game have been commented out.

### CODE REVIEW

**Review by rOluochKe — CHANGES_REQUESTED** [review]
Hello @JelilFaisalAbudu and @Albertino2020 
### Good effort you have put in :clap: However, you have a few issues you need to work on before your project could be approved.
### The logic is still off, you are required to have these functionalities in this milestone(2)
- [ ] game asks players for their names
- [ ] game tells which player turn it is
- [ ] game asks the player to select from available moves
- [ ] game informs player if the selected move is invalid
- [ ] game displays board after the player move
- [ ] game informs player if the selected move is a winning move
- [ ] game informs player if the selected move is a draw move
- [ ] game repeats all actions for the next player's move

### In the logic you have submitted, it lucks a number of behaviors
- it does not ask for my name
- it does not reject invalid data and ask me to put in a valid data
- I cannot win, lose or draw the game
- When I play, the board is not changing, accepting the inputs
game


### Make all these changes and submit for another code review. Good luck.
- Happy coding!

By Raymond
slack @iamraymond

**Comment by JelilFaisalAbudu** [comment]
# Milestone 2 2nd code Review Request
- All the required changes requested by the TSE in the previous code review have been effected and ready to for another review.
- Player can choose between letter "X" or "O" to play with.
- The game board is now displaying for each player's move.
- The validity of the player's move is added.

**Comment by lekegitrepo** [comment]
#### Hi @JelilFaisalAbudu and @Albertino2020 :clap: :clap:
#### Well done. However, the game is still the same as mentioned in the previous review.
#### Please, implement the required changes mentioned by the last reviewer and submit for another review.

### Reviewed by Temitayo Adeleke.

**Review by lekegitrepo — CHANGES_REQUESTED** [review]
#### Status: Request changes.

**Comment by JelilFaisalAbudu** [comment]
# Code Review Request
- The changes requested by the TSE before the current TSE's request have already been made.
- I think that there is a technical challenge with the system that was making my latest changes not to be reflected in the repository. But after cross checking, everything is in the respective branch(user_interface) from which I made the pull request.

File: bin/main.rb
Comment (shshamim63) [inline bin/main.rb]: [Optional]
There is another approach to declare it.
input_symb2 = input_symb1 == 'O' ? 'X' : 'O'

File: bin/main.rb
Comment (shshamim63) [inline bin/main.rb:37]: :warning: Comparing with a sudden number indicates that there is a problem based on scalability and feasibility. These are also the terms of single responsibility principles. I think you don't have the intension to use this condition on your game logic milestone.

**Review by shshamim63 — CHANGES_REQUESTED** [review]
## Status :racing_car: Required Changes :recycle: 
___
Hi @JelilFaisalAbudu and @Albertino2020 :wave: 
Almost covered all the changes asked by the previous reviewer. Just a simple issue is blocking me to approve it right now. Let's check it.
___
### :spades: Mandatory Changes :spades: 
___
- I can not see any message or comment related to winning status or a draw status. You don't to place all the logic in this milestone so a simple puts or comment is enough.
___
##### Please fix the above-mentioned issue and send another review request.
___
## :herb: Happy Coding :herb: 
### Shakhawat Hossain

**Comment by JelilFaisalAbudu** [comment]
# Review Request
- Comments have been added to show where check for winning status and also a draw status.

**Review by pelzolga123 — APPROVED** [review]
Approved.

**Comment by pelzolga123** [comment]
## Good job :+1: Project Approved.
You did all changes required previously. Go ahead and merge to develop.
_Goodluck in your next project_ *Happy coding*

**Pull request merged**
